JobRecommendationService.md

```diff
@@ -0,0 +1,173 @@
+|----------|------|----------|-------------|
+| `source_jobs` | List<[**Job**](#job)> | **TRUE** | A list of **Job**.
+
+Note: currently only the first job in the list is used for recommendations.
+
+#### Job
+| Param    | Type | Required | Description |
+|----------|------|----------|-------------|
+| `job_did` | String | **TRUE** | This String is required to start with `J` and have a length that is less than the given `MAX_LENGTH`, specifically 22 currently.
+
+### UserData
+| Param    | Type | Required | Description |
+|----------|------|----------|-------------|
+| `visitor_id` | String | **FALSE** | This starts with `VX` and is required to be shorter than 50 (`MAX_LENGTH`) characters .
+| `email_address` | String | **TRUE** | The email of the registered user.
+
+**UserData** contains two String parameters: `visitor_id` and `email_address`. However, currently only `email_address` is used to retrieve user applications.
+
+### UserPreferences
+| Param    | Type | Required | Description |
+|----------|------|----------|-------------|
+| `location_filter` | [**PreferredLocation**](#preferredlocation) | **FALSE** | Preferered working location.
+| `compensation_filter` | [**Compensation**](#compensation) | **FALSE** | Expected compensation for a job application or a user.
+| `carotene_titles` | List<[**CaroteneTitle**](#carotenetitle)> | **FALSE**<sup>[1](#myfootnote1)</sup> | A list of carotene job titles that the user is interested.
+| `carotene_version` | [**CaroteneVersion**](#caroteneversion) | **FALSE**<sup>[1](#myfootnote1)</sup> | The email of the registered user.
+
+**UserPreferences** provides additional constraints for recommendations. The service will rank recommendations from FAISS indexes and GBR data based on the location,  complesation and job titles. All the parameters are optional. But it has to be in valid format if any of them is provided.
+
+<a name="myfootnote1">1</a>: Carotene titles and version must be presented together or left blank, otherwise will be considered invalid.
+
+#### PreferredLocation
+| Param    | Type | Required | Description |
+|----------|------|----------|-------------|
+| `latitude` | double | **TRUE** | latitude of the location, range in `(-90, 90)`.
+| `longitude` | double | **TRUE** | longitude of the location, range in `(-180, 180)`.
+| `radius_in_miles` | float | **TRUE** | The redius of the location to specify the range of location filter, which has to be a positive number.
+
+**PreferredLocation** uses coordinates along with the radius to indicate the preferred work location. All the parameters have to be present in valid values.
+
+#### Compensation
+| Param    | Type | Required | Description |
+|----------|------|----------|-------------|
+| `type` | String | **TRUE** | Type of the compensation, currently only support `HOURLY` and `ANNUALLY`.
+| `amount` | double | **TRUE** | The total expected amount of the compensation, which is required to be a positive number.
```

Comment:
I would leave off the positive requirement. If people are passing in negative values on this field then they get the 400 they deserve imo.

Reply:
Alright, let me remove this. 
Actually, this haven't been checked yet. so I created this pr https://github.com/cbdr/job-recommendation-service/pull/32.